### PR TITLE
Resolve Dexcom Connection issues

### PR DESCRIPTION
**Pull request by darinkrauss**
This is a conglomeration of the following approved PRs:
- https://github.com/tidepool-org/platform/pull/732
- https://github.com/tidepool-org/platform/pull/733
- https://github.com/tidepool-org/platform/pull/734
- https://github.com/tidepool-org/platform/pull/735
- https://github.com/tidepool-org/platform/pull/736
- https://github.com/tidepool-org/platform/pull/752
- https://github.com/tidepool-org/platform/pull/755

The commits are exactly the same as with the above PRs, except the latest commit which is a merge from the latest `master` branch.

### CODE REVIEW

**Review by jh-bate — APPROVED** [review]
LGTM - just two queries

File: data/types/settings/cgm/level_alert_DEPRECATED.go [hunk 1]
```diff
@@ -10,12 +10,12 @@ import (
...
 	HighLevelAlertDEPRECATEDLevelMmolLMaximum float64 = 22.20299
-	HighLevelAlertDEPRECATEDLevelMmolLMinimum float64 = 6.66090
-	LowLevelAlertDEPRECATEDLevelMgdLMaximum   float64 = 100
+	HighLevelAlertDEPRECATEDLevelMmolLMinimum float64 = 5.55075
+	LowLevelAlertDEPRECATEDLevelMgdLMaximum   float64 = 150
 	LowLevelAlertDEPRECATEDLevelMgdLMinimum   float64 = 59
-	LowLevelAlertDEPRECATEDLevelMmolLMaximum  float64 = 5.55075
+	LowLevelAlertDEPRECATEDLevelMmolLMaximum  float64 = 8.32612
 	LowLevelAlertDEPRECATEDLevelMmolLMinimum  float64 = 3.27494
 )
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/types/settings/cgm/level_alert_DEPRECATED.go:20]: The mg/dL values are legit and what I've seen in production. The mmol/L values are simply calculated based upon the mg/dL values. (I have not had the opportunity to see what the OUS Dexcom API returns. Namely, I've only seen mg/dL values.)

File: platform/client.go [hunk 2]
```diff
@@ -25,7 +25,19 @@ type Client struct {
...
 
+// Modern platform-based services encode all errors with detailed information in the
+// response body. Deserialize those errors and pass onward.
 func NewClient(cfg *Config, authorizeAs AuthorizeAs) (*Client, error) {
+	return NewClientWithErrorResponseParser(cfg, authorizeAs, client.NewSerializableErrorResponseParser())
+}
+
+// Legacy services (shoreline, highwater, gatekeeper, etc.) do not serialize their errors
+// in the response body in a manner that platform can parse.
+func NewLegacyClient(cfg *Config, authorizeAs AuthorizeAs) (*Client, error) {
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply platform/client.go:36]: Yea, probably, but we don't do that for the rest of the code, so I'm hesitant to add now. The word "Legacy" in the function name is pretty clear.

**Review by toddkazakov — DISMISSED** [review]

**Review by toddkazakov — DISMISSED** [review]

**Review by jh-bate — APPROVED** [review]
LGTM

**Pull request merged**